### ISSUE TITLE
Displaying grouped bars with height 0 and , but not its label 

### ISSUE THREAD

**Issue by psandev** [has2k1/plotnine#408]
Hi,
here is my code, which creates a grouped bar chart with labels. 
p = ggplot(data5, aes(x='Illumination', fill='Repeat_Color')) +\
   geom_bar(position='dodge') +\
    ggtitle('Types Detection vs. Distance from Focus') +\
    labs(x='Illumination', y='Count') +\
    geom_text(aes(label='stat(count)'),
              position=position_dodge(width=0.9),
         stat='count',
         va='bottom' ) + \
    scale_y_continuous(expand=(0, 0, 0.1, 0)) +\
    facet_grid('Type~',scales='free_y') +\
    theme(
        text=element_text(size=14),
          legend_title_align='left',
          legend_box_margin=5,
        legend_text = element_text(size=12),
        axis_text_x=element_text(rotation=45, hjust=1)
         )

The resulting chart is this.
![image](https://user-images.githubusercontent.com/30928228/84581031-8e9d2200-ade5-11ea-8c3b-557cbc3b2020.png)

By using pandas.groupby I am almost able to get what I want:

![image](https://user-images.githubusercontent.com/30928228/84581064-e3d93380-ade5-11ea-82d5-160368e77dba.png)
My questions are:
1. Is it possible to get the chart by using just plotnine? GGplot2 allows it, but I haven't found a plotnine analog.
2. Is it possible not to present the labels with values 0 ?

**Comment by has2k1** [comment]
The proper solution should be `position_dodge(width=.., preserve='single')` but because of a bug it cannot deal with `geom_text`. The other solution is to use a function to "remove" the zeros.

```python
def remove_zeros(arr):
    return ['' if x == 0 else str(x) for x in arr]

(ggplot()
...
+ geom_text(aes(y='stat(count)', label='remove_zeros(stat(count))')
)
```

**Comment by psandev** [comment]
Thank you very much. Very nice trick. It does remove the zeros.
I also wanted not to remove the bars with zero height, but I guess it will
be solved when the bug in geom_text is fixed. Although, I saw similar
behavior in ggplot2.

preserve = "single" messes up geom_text in ggplot2 in R as well.

There were also some answers with stat='identity' in geom_bar, but it seems
to work only when the labels come from a data frame column, not from a stat
calculation. Like this:

geom_bar(colour = "black",position = "dodge",stat = "identity")

I got it from here:

https://stackoverflow.com/questions/10326729/dont-drop-zero-count-dodged-barplot

Can stat = 'identity' be used in geom_bar together with the custom
stat in geom_text?
